packages/mui-material/src/Typography/Typography.js

```diff
@@ -226,6 +226,7 @@ Typography.propTypes /* remove-proptypes */ = {
   /**
    * If `true`, the element will be a paragraph element.
    * @default false
+   * @deprecated use the `component` prop instead, or use variant="body2" to get the same behavior.
```

Comment:
`.d.ts` files are the source of truth. Every time you modify prop types in a .d.ts file you should run yo apply changes to the corresponding .js file.

```
pnpm proptypes
pnpm docs:api
```